Gems/PhysX/Code/Source/PhysXCharacters/Components/CharacterGameplayComponent.cpp

```diff
@@ -66,9 +66,9 @@ namespace PhysX
     {
     }
 
-    CharacterGameplayComponent::CharacterGameplayComponent(const CharacterGameplayConfiguration& config)
-        : m_gravityMultiplier(config.m_gravityMultiplier)
-        , m_groundDetectionBoxHeight(config.m_groundDetectionBoxHeight)
+    CharacterGameplayComponent::CharacterGameplayComponent(AZStd::unique_ptr<CharacterGameplayConfiguration> config)
+        : m_gravityMultiplier(config->m_gravityMultiplier)
+        , m_groundDetectionBoxHeight(config->m_groundDetectionBoxHeight)
```

Comment:
update: I don't think we want to update this interface if we can help it, passing a `unique_pointer` makes less sense to me here. What might be better is adding a `const` accessor to `CharacterGameplayComponent` to retrieve the current configuration if you need to check its state.

Reply:
I think CharacterControllerComponent is actually the one which should probably be updated (but that's out of scope for this work). It's still using the old pattern where we used to have shared pointers everywhere.

Reply:
+1 to Dave's point, I think if you can just change this back and everything works @FuzzyCarterAWS I'd leave as is. For a `const` accessor I just meant something like...

```
const CharacterGameplayConfiguration& GetCharacterGameplayConfiguration() const
{
    return m_characterGameplayConfiguration;
}
```

If we're not storing it we might not be able to do that exactly and would need to return the equivalent for `m_gravityMultiplier` and `m_groundDetectionBoxHeight` but if they aren't needed I wouldn't bother.

Reply:
Reverted and got things working.

Reply:
Sounds great @FuzzyCarterAWS, if you don't need one no need to add it. Thanks!